### ISSUE TITLE
Truncate table content with expand/collapse behavior

### ISSUE THREAD

**Issue by jillian-hammer** [CDCgov/prime-reportstream#9790]
**Is your feature request related to a problem? Please describe.**
On ReportStream, we often have long strings of text within our tables. Sometimes, truncating the text field is the easiest way to keep the table legible. However, we haven't designed or engineered a user-friendly truncation behavior yet, and rely on horizontally and/or vertically scrolling the table, which can be cumbersome to read. 

**Describe the solution you'd like**
A simple solution would be truncating long strings beyond a specific max height or character count with an ellipses. When the user hovers over the truncated field, they can click to expand the cell to reveal the full content, and then click again to collapse the content. See the [Figma prototype link](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?type=design&node-id=5520-40473&scaling=min-zoom&page-id=4875%3A37085&starting-point-node-id=5520%3A40473) as a sketch reference.

Screengrab recording below:
![truncation_prototype_20230615.gif](https://images.zenhubusercontent.com/639236212968fbce989b5d65/94bf47c9-08c7-4cff-92cb-df9e2fa647b0)

**Describe alternatives you've considered**
- Horizontal and/or vertical scrolling (implemented)
- Opening in a modal

**Additional context**
- Design needs to identify the max. height and/or max. character count # for engineering
- Design needs to clarify what expansion looks like, ie, does the content just extend downwards or all directions? Is there animation?

**Comment by jillian-hammer** [comment]
FYI @etanb added the truncation behavior as a separate, low priority feature request here! i threw together a [figma prototype](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?type=design&node-id=5520-40473&scaling=min-zoom&page-id=4875%3A37085&starting-point-node-id=5520%3A40473) to express what i have in mind and linked it in the issue as well. 

timeline: perhaps we can revisit this once all of the tables are replaced on the live site, and we can more accurately assess how often truncation would be necessary & reprioritize from there.

**Comment by chris-kuryak** [comment]
@etanb @jillian-hammer 
This seems like it will need more definition from a design perspective before Engr can take it on. So my question is:
- Does this need to be separated into two tickets? A design specification creation ticket, and an engineering implementation ticket?
- Is this something that engineering can ask pointed questions about and design can answer directly in this ticket?
- Or is this something best served as a "collaborative" ticket that you both put time into over the course of a sprint?

**Comment by jillian-hammer** [comment]
@etanb @chris-kuryak hooray!!

from my perspective, I think option #2 is what I had in mind. I can provide a spec to start from in the ticket, and then we can collaboratively tinker from there.

**Comment by etanb** [comment]
Yes, I only think a separate Design vs. Eng ticket is needed when the Design is:

1) a full, brand new feature
2) requires ~1 sprint+ of Design work
3) eng can't move forward at all without said design

For this instance, we have some general ideas going forwards, and just want to tinker with functionality. Can do everything from this ticket.

**Comment by KeithNava** [comment]
@jillian-hammer checking in on if this is still the solution we want going forward?
